### DISCUSSION TITLE
Code formatting via Husky

### DISCUSSION THREAD

**Discussion by ITZSHOAIB** [wevm/viem#1895] [Idea / Feature Request]
I think we should start using Husky to format our code while committing.

https://dev.to/ruppysuppy/automatically-format-your-code-on-git-commit-using-husky-eslint-prettier-in-9-minutes-45eg

**Comment by jxom** [comment]
We already have a precommit hook.

**Reply by ITZSHOAIB** [reply]
But in the codebase, there are a lot of files with unformatted codes. Is there any specific reason for that?

**Reply by jxom** [reply]
What do you mean? Everything should be formatted already (this is also validated in CI).
